PythonVisualizations/SimpleQueue.py

```diff
@@ -230,133 +264,127 @@ def display(self):
         
         #in all cases: disable buttons as necessary
         if self.nItems == self.size:
-            buttons[0].config(state = DISABLED)
-            buttons[1].config(state = DISABLED)
+            self.buttons[0].config(state = DISABLED)
+            self.buttons[1].config(state = DISABLED)
             
         if self.nItems == 0:
-            buttons[2].config(state = DISABLED)
-            buttons[3].config(state = DISABLED)
+            self.buttons[2].config(state = DISABLED)
+            self.buttons[3].config(state = DISABLED)
 
-def pause(pauseButton):
-    global waitVar
-    waitVar.set(True)
 
-    pauseButton['text'] = "Play"
-    pauseButton['command'] = lambda: onClick(play, pauseButton)
 
-    canvas.wait_variable(waitVar)
-
-def play(pauseButton):
-    global waitVar
-    waitVar.set(False)
-
-    pauseButton['text'] = 'Pause'
-    pauseButton['command'] = lambda: onClick(pause, pauseButton)
-
-def onClick(command, parameter = None):
-    cleanUp()
-    #disableButtons()
-    if parameter:
-        command(parameter)
-    else:
-        command()
-    #if command not in [pause, play]:
-     #   enableButtons()
-
-def cleanUp():
-    global cleanup
-    if len(cleanup) > 0:
-        for o in cleanup:
-            canvas.delete(o)
-    outputText.set('')
-    window.update()
+    def onClick(self,command, parameter = None):
+        self.cleanUp()
+        #disableButtons()
+        if parameter:
+            command(parameter)
+        else:
+            command()
+        #if command not in [pause, play]:
+         #   enableButtons()
 
 # Button functions
 
-def clickInsertRear():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            queue.insertRear(int(entered_text))
-        else:
-            outputText.set("Input value must be an integer from 0 to 99.")
-        textBox.delete(0, END )
+    def clickInsertRear(self):
+        entered_text = self.textEntries[0].get()
+        if entered_text:
+            val = int(entered_text)
+            if val < 100:
+                self.insertRear(int(entered_text))
+            else:
+                self.outputText.set("Input value must be an integer from 0 to 99.")
+            self.textEntries[0].delete(0, END )
         
-def clickInsertFront():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            queue.insertFront(int(entered_text))
-        else:
-            outputText.set("Input value must be an integer from 0 to 99.")
-        textBox.delete(0, END )
+    def clickInsertFront(self):
+        entered_text = self.textBox.get()
+        if entered_text:
+            val = int(entered_text)
+            if val < 100:
+                self.insertFront(int(entered_text))
+            else:
+                self.outputText.set("Input value must be an integer from 0 to 99.")
+            self.textBox.delete(0, END )
```

Comment:
This function fails every time because there is no `self.textBox` attribute.  Since we are only implementing the Queue and not the Deque, the easiest solution is to get rid of this function (and the associated Deque, Insert at Front, Deleted from End, and Queue operations)